solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -1450,18 +1445,20 @@ public void initializeMetrics(
                       .putAll(baseGaugeCoreAttributes)
                       .put(TYPE_ATTR, "start_time")
                       .build());
-            }));
+            })));
 
     if (coreContainer.isZooKeeperAware())
-      this.coreIsLeader =
+      observables.add(
           parentContext.observableLongGauge(
               "solr_core_is_leader",
               "Indicates whether this Solr core is currently the leader",
               (observableLongMeasurement -> {
                 observableLongMeasurement.record(
                     (coreDescriptor.getCloudDescriptor().isLeader()) ? 1 : 0,
                     baseGaugeCoreAttributes);
-              }));
+              })));
+
+    this.toClose = List.copyOf(observables);
```

Comment:
why copy it?  I could see maybe using an unmodifiable wrapper but at least right now there's no point.